### PR TITLE
Change "Version" to "PackageVersion" in Versions.props

### PR DESCRIPTION
**Pull request by mmitche**
We use PackageVersion in most places, so let's be consistent.  Also includes the following fixes:
- Add shouldn't pass paths to files explicitly
- Version.Details.xml should add the dependency node if it doesn't exist.
- Versions.props should add property groups where needed, should also update a package name property if possible.  This is not yet done for update.  Work is covered in https://github.com/dotnet/arcade/issues/1095 for general refactoring.
- Use local-name xpath queries so that projects that use a default xmlns (common for MSBuild) update correctly.

### CODE REVIEW

**Comment by mmitche** [comment]
Fixes #1140

**Comment by mmitche** [comment]
@weshaggard FYI

**Comment by jcagme** [comment]
Are "Version" and "PackageVersion" the only suffixes used in Version.props or we need to make this a bit more generic?

**Comment by mmitche** [comment]
@jcagme We should keep this consistent and use PackageVersion everywhere.  When I get around to generic updaters I may make it configurable but not sure.

**Comment by jcagme** [comment]
In this case we'll need to change Version.props around. At least https://github.com/dotnet/arcade/blob/master/eng/Versions.props uses Version. Haven't checked other repos to see what they use

**Comment by mmitche** [comment]
@jcagme Okay good call.  It will need to be changed around on Maesto rollout though.

**Comment by jcagme** [comment]
Or... could we add some logic here to update the values? With that we don't need to change anything manually. For example in Arcade's case, if `SystemIOPackagingVersion` is found we'll replace it with `SystemIOPackagingPackageVersion`? With this we'll also make sure `Version.props` in every repo is "compliant"

**Comment by weshaggard** [comment]
> For example in Arcade's case, if SystemIOPackagingVersion is found we'll replace it with SystemIOPackagingPackageVersion? With this we'll also make sure Version.props in every repo is "compliant"

You should be careful about this as it will now break all usages of that property in the repo. Perhaps a better transition is to support both patterns.

**Comment by mmitche** [comment]
@tmat Do you know of any places where we have:
- *Version form properties in Versions.props
- That are also updated by subscription.

I don't know of any of these, it would only be arcade subscriptions right now.  I'd rather not introduce kludge to support both, and just go change those places.

**Comment by weshaggard** [comment]
>I'd rather not introduce kludge to support both, and just go change those places.

I'd honestly plumb it in as an option and default it to PackageVersion and let folks override it if they want.

**Comment by mmitche** [comment]
@weshaggard I'll plumb it in when I generify the overall dependency management.  But I'd heavily bias towards keeping everything consistent.  IMO there's not a good reason to not have it be consistent, besides someone just not wanting to go replace a few of strings in proj files.

**Comment by weshaggard** [comment]
If we are just worried about consistency and corefx is the only one currently with "PackageVersion" I'm happy to update them to just "Version" if that is the pattern we think we want long term. At this point corefx only uses these properties in one place currently.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs [hunk 1]
```diff
@@ -179,42 +181,122 @@ void BuildDependencies(XmlNodeList dependencies)
...
+        ///     
+        ///     <!-- Package versions -->
+        ///     <PropertyGroup>
+        ///         <MicrosoftDotNetApiCompatPackageVersion>1.0.0-beta.18478.5</MicrosoftDotNetApiCompatPackageVersion>
+        ///     </PropertyGroup>
+        ///     
+        ///     See https://github.com/dotnet/arcade/blob/master/Documentation/DependencyDescriptionFormat.md for more
+        ///     information.
+        /// </summary>
+        /// <param name="repo">Path to Versions.props file</param>
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs]: NIT: Fix the parameter descriptions here.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs [hunk 2]
```diff
@@ -179,42 +181,122 @@ void BuildDependencies(XmlNodeList dependencies)
...
+        /// <param name="dependency">Dependency information to add.</param>
+        /// <returns>Async task.</returns>
+        public async Task AddDependencyToVersionsProps(string repo, string branch, DependencyDetail dependency)
         {
-            XmlDocument versionProps = await ReadVersionPropsAsync(filePath, null);
+            XmlDocument versionProps = await ReadVersionPropsAsync(repo, null);
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs]: Why is this not passing branch as well?

**Comment by mmitche** [comment]
> I'd rather not introduce kludge to support both, and just go change those places.
> 
> I'd honestly plumb it in as an option and default it to PackageVersion and let folks override it if they want.

I will, but it's not easily configurable right now.  That will go in the refactor, which will go as soon as I get proper tests in place for these files.

**Comment by tmat** [comment]
I'd definitely prefer `-Version` over `-Version`. All former RepoToolset repos use `-Version` and it's shorter.

**Comment by tmat** [comment]
Definitely +1 for consistency.

**Comment by mmitche** [comment]
@tmat All of core and CLI and aspnet currently use "PackageVersion:", and those that implemented ProdCon v1 probably use PackageVersion. Anyways, i'm not particularly tied to one or the other, we're just in an awkward spot while we don't have this latest code rolled out.

**Comment by tmat** [comment]
So this is gonna be a bit trickier. The `DefaultVersions.props` in Arcade uses `-Version`. If we switch to `-PackageVersion` for consistency we would need to update all repos that use Arcade, which might be 30+ at this point.

Could the tool recognize both suffixes for now? If it doesn't end with `-PackageVersion` try `-Version`? Seems like that shouldn't complicate things much. We can then transition repos gradually.

**Comment by tmat** [comment]
https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props#L53-L74

**Comment by mmitche** [comment]
@tmat Yeah I'll do that then.

**Comment by mmitche** [comment]
@weshaggard @tmat I found a few other bugs while writing tests for this, and my testing branch is based off of this one.  I'm gonna close this PR for now since wes is unblocked and check in tests+fixes for this as a whole.

**Pull request closed without merging**